packages/core-types/src/CoreDate.ts

```diff
@@ -138,17 +136,17 @@ export class CoreDate extends CoreSerializable {
     }
 
     private get asValidDateTime(): DateTime<true> {
-        if (!this.dateTime.isValid) throw new TransportError("The date is invalid.");
+        if (!this.dateTime.isValid) throw new Error("The date is invalid.");
```

Comment:
Would it make sense to have a CoreError here?

Reply:
no, I had this, but then removed it.

Reply:
To give more context. We already have a CoreError, but this has code, message, ... - this would be too much for this class.